mesh/mesh.cpp

```diff
@@ -8605,8 +8605,9 @@ int Mesh::FindPoints(DenseMatrix &point_mat, Array<int>& elem_ids,
             if (!tbf[k]) { continue; }
             Vector center(data+k*spaceDim,spaceDim);
             GetElementVertices(e_idx[k], vertices);
-            for (int v = 0; v < vertices.Size() || elem_ids[k] != -1; v++)
+            for (int v = 0; v < vertices.Size(); v++)
```

Comment:
This looks fine as a local replacement.

Alternatively, you could encode this check in the ``tbf`` ("to be found"?) array a few lines up.
E.g.
```cpp
// Line 8596
tbf[k] = ! outside && elem_ids[k] == -1;
```
Then ``elem_ids[k]`` can only change in a single place within the loop (line 8621), and that already has an early return.